### DISCUSSION TITLE
Why is the `Button` size in this case not its preferred size?

### DISCUSSION THREAD

**Discussion by Enyium** [slint-ui/slint#6252] [General] [answered]
I have [this](https://slintpad.com/?snippet=import+%7B+Button+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+Demo+%7B%0A++++width%3A+300px%3B%0A++++height%3A+300px%3B%0A%0A++++Rectangle+%7B%0A++++++++background%3A+lime%3B%0A%0A++++++++Rectangle+%7B%0A++++++++++++background%3A+fuchsia%3B%0A++++++++++++%2F%2Fwidth%3A+button.width%3B%0A++++++++++++%2F%2Fheight%3A+button.height%3B%0A++++++++++++width%3A+button.preferred-width%3B%0A++++++++++++height%3A+button.preferred-height%3B%0A%0A++++++++++++button+%3A%3D+Button+%7B%0A++++++++++++++++%2F%2Fwidth%3A+self.preferred-width%3B%0A++++++++++++++++%2F%2Fheight%3A+self.preferred-height%3B%0A++++++++++++++++text%3A+%22Foo%22%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=fluent-light) demo:

```slint
import { Button } from "std-widgets.slint";

export component Demo {
    width: 300px;
    height: 300px;

    Rectangle {
        background: lime;

        Rectangle {
            background: fuchsia;
            //width: button.width;
            //height: button.height;
            width: button.preferred-width;
            height: button.preferred-height;

            button := Button {
                //width: self.preferred-width;
                //height: self.preferred-height;
                text: "Foo";
            }
        }
    }
}
```

When you change around the styles a bit, the `Button` is rendered bigger than its enclosing `Rectangle`:

![button-1](https://github.com/user-attachments/assets/e2e29f6a-34fd-412a-89b2-f04463c998a9)
![button-2](https://github.com/user-attachments/assets/8540481c-40d9-4f12-b5d1-b203a0d64581)
![button-3](https://github.com/user-attachments/assets/8188b915-3b15-4f53-bf80-400714ac343b)

- Is that a bug, or does it work as intended?
- Who defines the `Button`'s size here?
- Could this be related to #6243?

**Comment by ogoffart** [comment, marked as answer]
It looks like the Button's min-height is bigger than its preferred-size.
It sounds like a bug, but not sure on what level.
The Button's specifies a minimum size. But the preferred size is automatically computed based on the contents.
When the width of the button is computed, its minimum size is taken into account, but the size of the fuchsia rectangle it taking the actual preferred-size because of the binding.

I'd say this is either a bug in the Widgets implementation as they don't specify a preferred size. Or the bug is that the auto-computation of the preferred size should take the explicit minimum size into account.

**Reply by Enyium** [reply]
Could you then please convert this discussion into an issue?

**Reply by ogoffart** [reply]
Github don't have the feature to convert discussion to issue. But I created issue #6265 from it.
